### ISSUE TITLE
let_var_whitespace rule is triggered when on one line before there is #warning statement

### ISSUE THREAD

**Issue by Igor-Palaguta** [realm/SwiftLint#2544]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

let_var_whitespace rule is triggered when on one line before there is #warning statement

```swift
struct SomeStruct {
    let first: Int
    let second: Int

    #warning("Warning")
    let third: Int
}
```

`Variable Declaration Whitespace Violation: Let and var should be separated from other statements by a blank line. (let_var_whitespace)`

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.29.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
SwiftLint.pkg
* Paste your configuration file:

```yml
opt_in_rules:
  - let_var_whitespace
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
No
* Which Xcode version are you using (check `xcode-select -p`)?
Version 10.0 (10A255)
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
struct SomeStruct {
    let first: Int
    let second: Int

    #warning("Warning")
    let third: Int
}
```

**Comment by Igor-Palaguta** [comment]
@marcelofabri yes sure, let me check

**Comment by Igor-Palaguta** [comment]
@undeaDD 

Checked rule implementation. And fixing it for `#warning` is one line fix https://github.com/realm/SwiftLint/pull/2629. But for typealiases not so fast. As we should check that same type is used. 

e.g.

Trigger:
```swift
typealias A = Int
typealias B = Float
let a = "Hello"
```

Not trigger:
```swift
typealias A = Int
let a: A
```

And not sure what to do here:
```swift
typealias A = Int
typealias B = Float
let a: B
```

**Comment by marcelofabri** [comment]
Fixed in #2629. 

@undeaDD Please file a new issue for what you've described.
